### PR TITLE
tweaks to the theory manual

### PR DESCRIPTION
**Pull request by JustinSGray**

### CODE REVIEW

File: openmdao/docs/theory_manual/total_derivs/coupled_vs_uncoupled.rst [hunk 1]
```diff
@@ -46,7 +46,7 @@ In other words, there will be non-zero entries both above and below the diagonal
     :align: center
     :width: 50%
 
-    The linear system for a coupled system must be solved using either a direct factorization or an iterative linear solver.
+    The linear solver for a coupled system must be solved using either a direct factorization or an iterative linear solver.
```
Comment (swryan) [inline openmdao/docs/theory_manual/total_derivs/coupled_vs_uncoupled.rst]: a solver must be solved?

File: openmdao/docs/theory_manual/total_derivs/coupled_vs_uncoupled.rst [hunk 2]
```diff
@@ -57,10 +57,19 @@ There are two basic categories of linear solver that can be used in this situati
...
 
+Which type of solver is best for your model use is heavily case-dependent and sometimes can be a difficult question to answer absolutely.
+However, there are a few rules of thumb that can be used to guide most cases:
+
+    #. direct solvers are very simple to use and for smaller problems is likely to be the best option.
+    The only downside is that the cost of computing the factorization scales with :math:`n^3` where :math:`n` is the length of your variable vector and so the compute cost can get out of control.
+    If :math:`n` < 2000 try this solver first.
+    #. iterative solvers are more difficult to use because they do not always succeed in finding a good solution to the linear problem.
+    Often times they require preconditioners in order to be effective.
+    However, with adequate preconditioning iterative solvers can dramatically out perform direct solvers for even moderate sized problems.
```
Comment (swryan) [inline openmdao/docs/theory_manual/total_derivs/coupled_vs_uncoupled.rst]: `outperform`

File: openmdao/docs/theory_manual/total_derivs/index.rst [hunk 3]
```diff
@@ -5,6 +5,15 @@ How Total Derivatives are Computed
 **********************************
 
 This is a comprehensive document about how OpenMDAO solves for total derivatives.
+Total derivatives are primarily needed for gradient based optimizations methods.
+While it is possible to use finite-differences to approximate total derivatives, for larger multidisciplinary models this approach is notoriously inaccurate.
+Using OpenMDAO's total derivatives features can significantly improve the efficiency of your gradient based optimization implementation.
+
+.. note::
+
+    total derivatives are also useful for other applications such as gradient enhanced surrogate modeling and dimensionality reduction for active subspaces
```
Comment (swryan) [inline openmdao/docs/theory_manual/total_derivs/index.rst]: capitalize and end with a period?

File: openmdao/docs/theory_manual/total_derivs/index.rst [hunk 4]
```diff
@@ -24,27 +33,39 @@ Since we are focused on total derivatives, we will assume that the partial deriv
...
 -----------------------------
 
-Total derivative calculations in OpenMDAO are based on the `Unified Derivative Equations`_
-These equations allow you to compute total derivatives across any system of nonlinear equations, by solving a linear system in the forward (direct) form:
+A model's fundamental purpose is to compute an objective or constraint as a function of design variables.
+In order to perform those computations the model moves data through many different calculations (defined in the components) using
+many intermediate variables.
+Internally, OpenMDAO doesn't distinguish between objective/constraint variables, design variables, or intermediate variables;
+they are all just variables, following the mathematical formulation prescribed by the `MAUD architecture`_, developed by Hwang and Martins.
+Using that formulation, it is possible to compute the total derivative any variable with respect to any other variable by solving a linear system of equations, called the `Unified Derivative Equations`_ (UDE).
```
Comment (swryan) [inline openmdao/docs/theory_manual/total_derivs/index.rst]: total derivative *of* any variable

File: release_notes.txt [hunk 5]
```diff
@@ -1,3 +1,73 @@
...
+    - MetaModelStructured -> MetaModelStructuredComp
+    - MetaModelUnStructured -> MetaModelUnStructuredComp
+    - MultiFiMetaModelUnStructured -> MultiFiMetaModelUnStructuredComp
+    - ExternalCode -> ExternalCodeComp
+
+- Driver metadata/options can now be set via init args (can still be set as options as well).
+- User no longer needs to specify PETSCvector for parallel cases (the framework figures it out now).
+
+- check_totals and check_partials no longer support the `supress_output` argument. Use `out_stream=None` instead.
+
```
Comment (Kenneth-T-Moore) [inline release_notes.txt]: Also in `check_partials`, "comps" was removed and replaced by "inlucdes" and "excludes", which allow glob patterns.

File: openmdao/docs/features/debugging/om_command.rst [hunk 6]
```diff
@@ -215,6 +215,6 @@ Using Commands under MPI
 In general, usage of openmdao subcommands under MPI is the same as usual, except the command will
 be preceded by `mpirun -n <num_procs>`.  For example:
 
-.. embed-shell-cmd::
-    :cmd: mpirun -n 2 openmdao summary multipoint_beam_opt.py
-    :dir: ../test_suite/scripts
+.. .. embed-shell-cmd::
+..     :cmd: mpirun -n 2 openmdao summary multipoint_beam_opt.py
+..     :dir: ../test_suite/scripts
```
Comment (swryan) [inline openmdao/docs/features/debugging/om_command.rst]: so this is broken?  noticed you had un-commented in previous commit. just making sure this is intentional

Comment (kmarsteller) [reply openmdao/docs/features/debugging/om_command.rst]: @jubs, have you seen this (above) question?

Comment (kmarsteller) [reply openmdao/docs/features/debugging/om_command.rst]: errr @JustinSGray

Comment (kmarsteller) [reply openmdao/docs/features/debugging/om_command.rst]: (If it is broken and stays commented out, should probably remove "For example:" above it, so it doesn't say, "For example: NOTHING" )

File: release_notes.txt [hunk 7]
```diff
@@ -1,3 +1,75 @@
+################################################################################################
+Release Notes for OpenMDAO 2.3.0
+
+May 14, 2018
```
Comment (kmarsteller) [inline release_notes.txt]: At the last minute, we will need to change this date to June...12?

File: release_notes.txt [hunk 8]
```diff
@@ -1,3 +1,75 @@
+################################################################################################
+Release Notes for OpenMDAO 2.3.0
+
+June 12, 2018
+
+New Features:
+--------------
```
Comment (kmarsteller) [inline release_notes.txt]: https://www.pivotaltracker.com/story/show/157854109 User can provide a CSV file to the DOEDriver to run an arbitrary set of cases.

Comment (kmarsteller) [reply release_notes.txt]: https://www.pivotaltracker.com/story/show/158011209 User can set check_partial_options to use CS if alloc=complex, but the component will do FD for check_partials of CS not available.

Comment (kmarsteller) [reply release_notes.txt]: https://www.pivotaltracker.com/story/show/149861856  User can give a case-prefix when the call run_model or run_driver by passing an argument to the function.

Comment (kmarsteller) [reply release_notes.txt]: https://www.pivotaltracker.com/story/show/157854243 should mention option change to 'procs_per_model' from 'parallel'

Comment (kmarsteller) [reply release_notes.txt]: https://www.pivotaltracker.com/story/show/157128289 SimpleGA driver can support models that need more than 1 proc

Comment (kmarsteller) [reply release_notes.txt]: https://www.pivotaltracker.com/story/show/157752792 Beam Example expanded/improved.

File: release_notes.txt [hunk 9]
```diff
@@ -33,6 +33,7 @@ New Features:
     - DotProductComp
     - MatrixVectorProductComp
     - ExternalCodeImplicitComp
+    - EQConstraintsComp
```
Comment (swryan) [inline release_notes.txt]: *singular*  :)

File: openmdao/docs/features/debugging/om_command.rst [hunk 10]
```diff
@@ -183,7 +183,7 @@ methods corresponding to each object instance that they were called on.  For mor
 
 
 .. _om-command-iprof-totals:
-
+..
```
Comment (swryan) [inline openmdao/docs/features/debugging/om_command.rst]: this looks accidental?

**Pull request merged**